tests/e2eTestForBoost/step-definitions/boost.steps.js

```diff
@@ -1,39 +1,31 @@
 const {Given, When, Then} = require('@cucumber/cucumber')
 let helpers = require('../../../helpers/common-actions');
-const assert = require('assert');
 
 
-Given(/^Login with (.*) and (.*)$/, async (username, password) => {
-   await boostPostPage.performLogin(username, password)
+Given(/^I login with (.*) and (.*)$/, async (username, password) => {
+   await boostPostPage.login(username, password)
 })
-
-Then (/^Create a text post$/,async function (){
-    await myFeedPage.CreatePost()
-    await helpers.waitForTimeout(page,5000);
-    await  myFeedPage.CreatePostFunc();
+Then(/^I create a text post$/,async function (){
+    await myFeedPage.createPost();
+    await helpers.waitForTimeout(page,10000);
+    await myFeedPage.validateCreatedPost();
 })
-
-Then (/^I click the boost button$/,async function (){
+Then(/^I click the Boost button$/,async function (){
     await myFeedPage.clickBoost();
 })
-
-When (/^I check the Boost Campaign Details$/,async function (){
-    await boostPage.VerifySetBudgetAndDuration();
+When(/^I check the Boost Campaign details$/,async function (){
+    await boostPage.verifySetBudgetAndDuration();
     await boostPage.setBudgetAndDuration();
 })
-
-When (/^Complete the Boost payment process$/,async function (){
+When(/^I complete the Boost payment process$/,async function (){
     await helpers.waitForTimeout(page,5000);
```

Comment:
Hey @aabrosimov-bl , i observed one thing for UI automation unlike API response which has a stable response time unless the API is heavy , the tests are very flaky... it needs some timeout for some tests otherwise its too fast, resulting in failure, Ive executed a couple of times & its uncertain when it passes/fails, so I added a delay where the tests were failing more often.

Reply:
In the future we should look for UI cues to indicate certain events instead of just using delays, because constant delays are always at a risk of being insufficient during service slowdowns etc., but that would be much more complex, so for now it's OK.